onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder_helpers.cc

```diff
@@ -235,7 +301,8 @@ Status BuildBatchMatMul(ModelBuilder& model_builder, const NodeUnit& node_unit)
   {
     const std::string b_new_perm = model_builder.GetUniqueName(b + "/new_perm"),
                       b_transposed = model_builder.GetUniqueName(b + "/transposed");
-    ORT_RETURN_IF_ERROR(AddNnapiTranspose(model_builder, gemm_b_inputs.front(), b_new_perm, {0, 2, 1}, b_transposed));
+    std::vector<int32_t> perm_data{0, 2, 1};
+    ORT_RETURN_IF_ERROR(AddNnapiTranspose(model_builder, gemm_b_inputs.front(), b_new_perm, gsl::make_span(perm_data), b_transposed));
```

Comment:
```suggestion
    ORT_RETURN_IF_ERROR(AddNnapiTranspose(model_builder, gemm_b_inputs.front(), b_new_perm, AsSpan<int32_t>({0, 2, 1}), b_transposed));
```
AsSpan is from core/common/span_utils.h. more info in the coding standards doc: https://github.com/microsoft/onnxruntime/blob/main/docs/Coding_Conventions_and_Standards.md